internal/rbd/rbd_journal.go

```diff
@@ -433,6 +433,7 @@ func updateTopologyConstraints(rbdVol *rbdVolume, rbdSnap *rbdSnapshot) error {
 		if rbdVol.Topology != nil {
 			rbdVol.Pool = poolName
 			rbdVol.DataPool = dataPoolName
+			rbdVol.JournalPool = poolName
```

Comment:
I think setting here seems to be wrong because, we need JournalPool to check if the image [exists](https://github.com/ceph/ceph-csi/blob/a021457ba1b759a68f5cced0777daa81fc3ab856/internal/rbd/rbd_journal.go#L263-L269)  This is called before reserveVol which internally calls this function, in this case as the JournalPool is not set we will never be able to get the metadata from the radosOmap

Reply:
can I validate it manually somehow in a local setup, i believe it is updating the value before at `parseVolCreateRequest` too

Reply:
you need to simulate that case where you need to panic after creating everything and before returning RPC response

Reply:
Updated the check i believe now I believe we wont have complaints `updateTopologyConstraints`